### DISCUSSION TITLE
How to properly stop worker on error

### DISCUSSION THREAD

**Discussion by VojtechBuba** [roadrunner-server/roadrunner#1899] [Q&A] [answered]
I want to ask, how to properly stop worker on error? In documentation https://docs.roadrunner.dev/php-worker/developer#stop-command there is example, where method `->stop()` on worker is used. It is followed by return statement. If i look at this method in interface `\Spiral\RoadRunner\WorkerInterface` I see comment 

>Attention, you MUST use continue; after invoking this method to let RoadRunner to properly stop worker. 

Should i use return after stop command or not?

**Comment by rustatian** [comment, marked as answer]
Hey @VojtechBuba 👋 
You may just exit the `while` loop. 
Continue needed to properly handle the internal signal from RR to gracefully stop that worker. After you send a stop, RR sends a SIGINT to the worker. So, if the worker is already dead (you used a return), RR will spam you an additional log, that it can't gracefully stop that worker. So, that is up to you. Anyway, RR will recreate the worker without any problems.

**Reply by VojtechBuba** [reply]
Than you for a quick reply.
